deeplabcut/pose_estimation_pytorch/apis/evaluate.py

```diff
@@ -167,6 +177,196 @@ def evaluate(
     return results, predictions
 
 
+def visualize_coco_predictions(
+    predictions: dict,
+    num_samples: int = 1,
+    test_file_json: str | Path = "test.json",
+    output_dir: str | Path | None = None,
+    draw_skeleton: bool = True,
+) -> None:
+    """
+    Visualize predictions using DeepLabCut's plot_gt_and_predictions function
+
+    Args:
+        predictions: Dictionary with image paths as keys and prediction data as values.
+                    Each prediction contains:
+                    - bodyparts: numpy array of shape (1, 37, 3)
+                    - bboxes: numpy array of shape (1, 4)
+                    - bbox_scores: numpy array of shape (1,)
+        num_samples: Number of samples to visualize
+        test_file_json: Path to test set JSON file
+        output_dir: Directory to save visualization outputs. If None, will create
+                   a directory next to test_file_json
+        draw_skeleton: Whether to draw skeleton connections between keypoints
+    """
+    # Load ground truth data
+    with open(test_file_json, "r") as f:
+        ground_truth = json.load(f)
+
+    if output_dir is None:
+        output_dir = os.path.join(
+            os.path.dirname(test_file_json), "predictions_visualizations"
+        )
+    os.makedirs(output_dir, exist_ok=True)
```

Comment:
For consistency, I think it would be better to continue using `pathlib` instead of `os.path` here:

```python
    if output_dir is None:
        output_dir = Path(test_file_json).parent / "predictions_visualizations"
    else:
        output_dir = Path(output_dir)

    output_dir.mkdir(exist_ok=True)
```